internal/cmd/common.go

```diff
@@ -321,12 +321,18 @@ func getCommandExportExtractMatches(lines []string) []CommandExportExtractMatch
 		for _, match := range test.FindAllStringSubmatch(line, -1) {
 			e := match[0]
 
-			parts := strings.Split(strings.TrimSpace(e)[len("export "):], "=")
+			parts := strings.SplitN(strings.TrimSpace(e)[len("export "):], "=", 2)
 			if len(parts) == 0 {
 				continue
 			}
 			key := parts[0]
 			ph := parts[1]
+
+			isExecValue := strings.HasPrefix(ph, "$(") || strings.HasPrefix(ph, ")")
+			if isExecValue {
+				continue
+			}
+
 			hasStringValue := strings.HasPrefix(ph, "\"") || strings.HasPrefix(ph, "'")
 			placeHolder := ph
 			if hasStringValue {
```

Comment:
First, you should put the `isExecValue` test after the test for a string value, since we can have `"$(command)"`.

Second, the test `strings.HasPrefix(ph, ")")` doesn't make sense - it would be better I think to just remove that test, and test only for the `$(` at the beginning.

Finally, you should test against `strings.TrimSpace(placeHolder)`, since there could be some leading spaces.

Reply:
@mxsdev So basically it is fine to have it first because if the command starts like `"$(command)"` it won't apply to this condition  `$(` != `"$(`

Reply:
Leading spaces could be a problem :/

Reply:
Discussed this offline, `isExecValue` position is totally fine! I misunderstood the original problem.